tools/autograd/templates/Functions.cpp

```diff
@@ -8,6 +8,12 @@ using at::IntList;
 
 namespace torch { namespace autograd {
 
+namespace {
+
+Tensor not_implemented(const char* name) {
+  throw std::runtime_error(std::string(name) + " is not differentiable");
```

Comment:
I think this error message is a bit misleading, because the operations might be differentiable but the derivatives were just not yet implemented.